geode-deployment-legacy/src/main/java/org/apache/geode/deployment/impl/legacy/LegacyJarDeploymentService.java

```diff
@@ -0,0 +1,191 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.geode.deployment.impl.legacy;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Path;
+import java.time.Instant;
+import java.util.Collections;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.stream.Collectors;
+
+import org.apache.logging.log4j.Logger;
+
+import org.apache.geode.annotations.Experimental;
+import org.apache.geode.deployment.JarDeploymentService;
+import org.apache.geode.internal.DeployedJar;
+import org.apache.geode.internal.JarDeployer;
+import org.apache.geode.logging.internal.log4j.api.LogService;
+import org.apache.geode.management.configuration.Deployment;
+import org.apache.geode.services.result.ServiceResult;
+import org.apache.geode.services.result.impl.Failure;
+import org.apache.geode.services.result.impl.Success;
+
+/**
+ * Implementation of {@link JarDeploymentService} that encpsulates the standard method of deploying
+ * jars by delegating to the {@link JarDeployer}.
+ *
+ * @since Geode 1.15
+ */
+@Experimental
+public class LegacyJarDeploymentService implements JarDeploymentService {
+
+  private static final Logger logger = LogService.getLogger();
+
+  private final Map<String, Deployment> deployments;
+  private JarDeployer jarDeployer;
```

Comment:
Just scanning over this class, it looks like `jarDeployer` can probably be `final`.

Reply:
Given the way the legacy works, this JarDeployer can be replaced with another JarDeployer. Once again, we are just wrapping the existing legacy approach with a service interface. We aren't changing how the existing code works